deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -372,4 +393,122 @@ public boolean cleanUp() {
+		}catch(Exception e) {
+			LOGGER.info("service is not statefulset");
+		}
+		if (v1StatefulSet!=null) {
+			latestPodSelector=getV1StatefulSetHandler().getPodSelector(v1StatefulSet.getStatus(), selector);
+			replicas = getV1StatefulSetReplicas(v1StatefulSet);
+		} else {
+			V1Deployment v1Deployment=getV1DeploymentHandler().get(apiClient, serviceName, namespace);
+			latestPodSelector = getV1DeploymentHandler().getPodSelector(apiClient, selector, true, namespace);
+			replicas = getV1DeploymentReplicas(v1Deployment);
+		}
+	
+		Watch<V1Pod> watch = null;
+		String namespaceSelector=ResourceSelectorUtil.getNamespaceSelector(namespace); 
+
+		Set<String> containersReadyPods = new HashSet<String>();
+		Set<String> readyPods = new HashSet<String>();
+		Set<String> initPods = new HashSet<String>();
+		Set<String> podScheduledPods = new HashSet<String>();
+		
+		ActivityContext readyActivityContext = null;
+		ActivityContext initializedActivityContext = null;
+		ActivityContext podScheduledActivityContext = null;
+
+		watch = Watch.createWatch(apiClient,
+				api.listPodForAllNamespacesCall(null, null,namespaceSelector , latestPodSelector, null, null, null, replicas*10, Boolean.TRUE, null),
+				new TypeToken<Watch.Response<V1Pod>>() {
+				}.getType());
+		try {
+			initializedActivityContext = new ActivityContext(DeployerActivity.POD_INITIALIZED);
+			WorkflowLogger.startActivity(initializedActivityContext);
+			
+			for (Watch.Response<V1Pod> item : watch) {
+				if (item.object.getStatus().getContainerStatuses().stream()
+						.anyMatch(s -> s.getRestartCount() > deployerConfig.getMaxPodRestartCount())) {
+					WorkflowLogger.endActivity(Status.FAILED);
+					throw new HyscaleException(DeployerErrorCodes.FAILED_TO_INITIALIZE_POD);
+				}
+				for (V1PodCondition v1PodCondition : item.object.getStatus().getConditions()) {
+					if (v1PodCondition.getType().equals(READY)) {
+						readyPods.add(item.object.getMetadata().getName());
+					} else if (v1PodCondition.getType().equals(INITIALIZED)) {
+						initPods.add(item.object.getMetadata().getName());
+					}else if (v1PodCondition.getType().equals(PODSCHEDULED)) {
+						podScheduledPods.add(item.object.getMetadata().getName());
+					} else {
+						for (V1ContainerStatus v1conContainerStatus : item.object.getStatus().getContainerStatuses()) {
+							String reason = null;
+							if (v1conContainerStatus.getState().getWaiting() != null) {
+								reason = v1conContainerStatus.getState().getWaiting().getReason();
+							}
+							if (K8SRuntimeConstants.POD_ERROR_SATES.contains(reason)) {
+								WorkflowLogger.endActivity(Status.FAILED);
+								throw new HyscaleException(DeployerErrorCodes.FAILED_TO_INITIALIZE_POD);
+							}
+						}
+					}
+				}
+				if (containersReadyPods.size() == replicas) {
+					break;
+				}
+			}
+			
+			if (initPods.size() == replicas) {
+				WorkflowLogger.endActivity(initializedActivityContext, Status.DONE);
```

Comment:
For Every Activity message we are printing we have to end it either with Done or Failed.